MapboxNavigation/NavigationViewController.swift

```diff
@@ -368,6 +394,19 @@ public class NavigationViewController: NavigationPulleyViewController, RouteMapV
         if routeProgress.currentLegProgress.alertUserLevel == .arrive {
             navigationDelegate?.navigationViewController?(self, didArriveAt: destination)
         }
+        
+        updateStyleIfNeeded()
+    }
+    
+    func updateStyleIfNeeded() {
+        if let styles = styles, styles != [styleForTimeOfDayAndBrightness] {
+            self.styles = [styleForTimeOfDayAndBrightness]
+            applyStyle()
```

Comment:
I think it's because appearance proxy only sets the properties when the view is added to a window, it doesn't work in real-time. Here's a hack that forces a refresh but it needs to be adapted because the map view crashes if it looses its `UIWindow` when it's active.
```swift
func refreshAppearance() {
        for window in UIApplication.shared.windows {
            for view in window.subviews {
                view.removeFromSuperview()
                window.addSubview(view)
            }
        }
    }
```

Try removing and adding the maneuver view in the same drawing cycle.